src/Symfony/Component/Serializer/Normalizer/BackedEnumNormalizer.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Normalizer;
+
+use Symfony\Component\Serializer\Exception\InvalidArgumentException;
+use Symfony\Component\Serializer\Exception\NotNormalizableValueException;
+
+/**
+ * Normalizes a {@see \BackedEnum} enumeration to a string or an integer.
+ *
+ * @author Alexandre Daubois <alex.daubois@gmail.com>
+ */
+final class BackedEnumNormalizer implements NormalizerInterface, DenormalizerInterface, CacheableSupportsMethodInterface
+{
+    /**
+     * {@inheritdoc}
+     *
+     * @throws InvalidArgumentException
+     *
+     * @return int|string
+     */
+    public function normalize($object, string $format = null, array $context = [])
+    {
+        if (!$object instanceof \BackedEnum) {
+            throw new InvalidArgumentException('The data must belong to a backed enumeration.');
+        }
+
+        return $object->value;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supportsNormalization($data, string $format = null)
+    {
+        return $data instanceof \BackedEnum;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws NotNormalizableValueException
+     */
+    public function denormalize($data, string $type, string $format = null, array $context = [])
+    {
+        if (!is_subclass_of($type, '\BackedEnum')) {
+            throw new InvalidArgumentException('The data must belong to a backed enumeration.');
+        }
+
+        if (!\is_int($data) && !\is_string($data)) {
+            throw new NotNormalizableValueException('The data is neither an integer nor a string, you should pass an integer or a string that can be parsed as an enumeration case of type '.$type.'.');
+        }
+
+        try {
+            return $type::from($data);
+        } catch (\Exception | \ValueError $e) {
```

Comment:
What kind of exception do we expect here? `\Exception` is a bit too generic imho.

Reply:
Updated. There are two cases here:

1. The value could not be found in the `BackedEnum`, which leads to a `ValueError`
2. If for any reason someones implements a `BackedEnum` polyfill without the `::from` method, it leads to an `Error`

The updated bit of code:

```php
try {
    return $type::from($data);
} catch (\Error $e) {
    throw new NotNormalizableValueException($e->getMessage(), $e->getCode(), $e);
}
```

Reply:
I would only implement the first case, tbh. I don't see a point in taking guesses about how broken a potential polyfill might be. Let's just catch the `ValueError`.

Reply:
Alright sounds good to me, fixed!